### PR TITLE
Use leak canary only for debug builds

### PR DESCRIPTION
**Pull request by maskaravivek**

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -51,9 +51,7 @@ dependencies {
     androidTestImplementation "com.android.support:support-annotations:${project.supportLibVersion}"
     androidTestImplementation 'com.android.support.test.espresso:espresso-core:3.0.1'
 
-    debugImplementation 'com.squareup.leakcanary:leakcanary-android:1.5.1'
-    releaseImplementation 'com.squareup.leakcanary:leakcanary-android-no-op:1.5.1'
-    testImplementation 'com.squareup.leakcanary:leakcanary-android-no-op:1.5.1'
+    implementation 'com.squareup.leakcanary:leakcanary-android:1.5.1'
```
Comment (whym) [inline app/build.gradle]: What is the purpose of this change? It appears to make the release build larger with the leakcanary classes which are not used.

File: app/src/main/java/fr/free/nrw/commons/CommonsApplication.java [hunk 2]
```diff
@@ -140,11 +140,14 @@ public void onCreate() {
         cacheData  = new CacheController();
     }
 
-    protected RefWatcher setupLeakCanary() {
+    protected void setupLeakCanary() {
```
Comment (whym) [inline app/src/main/java/fr/free/nrw/commons/CommonsApplication.java]: If the signature of this is to be changed, TestCommonsApplication's counterpart would need to be changed as well.

Comment (whym) [reply app/src/main/java/fr/free/nrw/commons/CommonsApplication.java]: By the way, having a RefWatcher instance could be useful, although we don't make use of it right now. : https://github.com/square/leakcanary/wiki/FAQ#how-do-i-use-it

**Comment by maskaravivek** [comment]
Closing this PR. It was an immature PR without exactly understanding leakcanary.

**Comment by whym** [comment]
To be honest I also had to read more about LeakCanary after leaving comments here.

For the record it appears that isInAnalyzerProcess() combined with the no-op dependency works similarly to the 'BuildConfig.DEBUG' flag making the latter redundant here. (I believe maskaravivek reached the same conclusion.)
